python/nano/src/bigdl/nano/quantization/quantization_inc.py

```diff
@@ -60,27 +64,61 @@ def __init__(self,
         :param outputs:     For tensorflow to specify names of outputs. e.g. outputs=['logits',]
         """
-        if conf:
-            qconf = Quantization_Conf(conf)
-        else:
-            qconf = Quantization_Conf('')
-            cfg = qconf.usr_cfg
-            # Override default config
-            cfg.model.framework = framework
-            cfg.quantization.approach = approach
-            cfg.tuning.strategy.name = tuning_strategy
-            if accuracy_criterion:
-                cfg.tuning.accuracy_criterion = accuracy_criterion
-            cfg.tuning.exit_policy.timeout = timeout
-            cfg.tuning.exit_policy.max_trials = max_trials
-            cfg.model.inputs = inputs
-            cfg.model.outputs = outputs
+        qconf = Quantization_Conf(conf)
+        cfg = qconf.usr_cfg
+        # Override default config
+        cfg.model.framework = framework
+        cfg.quantization.approach = approach
+        cfg.tuning.strategy.name = tuning_strategy
+        if accuracy_criterion:
+            cfg.tuning.accuracy_criterion = accuracy_criterion
+        cfg.tuning.exit_policy.timeout = timeout
+        cfg.tuning.exit_policy.max_trials = max_trials
+        cfg.model.inputs = inputs
+        cfg.model.outputs = outputs
         super().__init__(qconf)
 
+    def post_training_quantize(self, model, calib_dataloader=None, val_dataloader=None,
+                               metric=None):
+        self.check(calib_dataloader, val_dataloader, metric)
+        self.model = model
+        if calib_dataloader:
+            self.calib_dataloader = calib_dataloader
+        if val_dataloader:
+            self.eval_dataloader = val_dataloader
+            if metric.metric:
+                self.metric = Metric(metric, name='INC_' + type(metric).__name__)
+
+        quantized = self()
+        if quantized:
+            return quantized
+        else:
+            raise RuntimeError("Found no quantized model satisfying accuracy criterion.")
+
+    def check(self, calib_dataloader, val_dataloader, metric):
+        """
+        Call before self.__call__() to check if the object is well-initialized
+        for quantization.
+        """
+        if self.cfg.quantization.approach == 'post_training_static_quant':
+            assert calib_dataloader, \
+                "calib_calib_dataloader must not be None when approach is " \
+                "post-training static quantization."
+
+        if self.cfg.quantization.approach == 'post_training_dynamic_quant':
+            assert calib_dataloader is None, \
+                "calib_calib_dataloader must be None when approach is " \
+                "post-training dynamic quantization."
+
+        if metric.metric and not val_dataloader:
+            raise RuntimeError("val_dataloader must be specified when metric is not None.")
 
-class TorchMetricForINC:
-    def __init__(self, metric):
-        self.metric = metric
+
+class INCMetric(ABC):
+    metric: Any = None
```

Comment:
Why use a class variable? I think each instance of INCMetric should has its own metric variable.

Reply:
I agree, I tried to initiate a `INCMetric` by 
```python
INCMetric(metric=metric)
```
But in tensorflow, this will cause an error when dumping the configuration to yaml. So here I was intended to remove a tf or torch metric as attribute to avoid being dumped...